gufe/storage/resultclient.py

```diff
@@ -92,11 +103,127 @@ def __init__(self, external_store):
     def delete(self, location):
         self._result_server.delete(location)
 
-    def store_protocol_dag_result(self, result):
+    def store_result(self, result):
         # I don't know how we get the path information for the protocol dag
         # results
         self.result_server.store(...)
 
+    @staticmethod
+    def _gufe_key_to_storage_key(prefix, key):
+        """Create the storage key from the gufe key"""
+        prefix = prefix.split('/')  # remove this if we switch to tuples
+        cls, token = key.split('-')
+        tup = tuple(list(prefix) + [cls, f"{token}.json"])
+        # right now we're using strings, but we've talked about switching
+        # that to tuples
+        return "/".join(tup)
+
+    def _store_gufe_tokenizable(self, prefix, obj):
+        """generic function for deduplicating/storing a GufeTokenizable"""
+        for o in get_all_gufe_objs(obj):
+            key = self._gufe_key_to_storage_key(prefix, o.key)
+
+            # we trust that if we get the same key, it's the same object, so
+            # we only store on keys that we don't already know
+            if key not in self.result_server:
+                data = json.dumps(o.to_keyed_dict(),
+                                  sort_keys=True).encode('utf-8')
+                self.result_server.store_bytes(key, data)
+
+    def store_transformation(self, transformation):
```

Comment:
is store returning anything?  and/or should it return anything, what might be useful?  A copy of the gufekey it got stored as? (although this is already an attribute I guess)

Reply:
There's no additional information to be given by returning anything, although I'm not opposed to it returning something if there's value in it.